UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -608,8 +598,9 @@ void UpdateSensors()
         /// during inference the Sensors are used to write directly to the Tensor data. This will likely change in the
         /// future to be controlled by the type of brain being used.
         /// </summary>
-        public void GenerateSensorData()
+        public static List<Observation> GenerateSensorData(List<ISensor> sensors, float[] buffer, WriteAdapter adapter)
         {
+            var observations = new List<Observation>();
```

Comment:
Do you want to reduce the allocations by making GenerateSensorData() take an output List<Observation> parameter? Then RpcCommunicator can save it and reuse it.

Reply:
Done